File: java/src/main/java/challenges/aoc2024/day5/Application.java
package challenges.aoc2024.day5;

import challenges.aoc2024.day5.service.DayFiveSolver;
import challenges.aoc2024.day5.service.PageProductionService;
import service.TaskSolver;

public class Application {
    public static void main(String[] args) {
        final TaskSolver solver = new TaskSolver(new DayFiveSolver(new PageProductionService()));
        solver.solveTasks();
    }
}

File: scala/src/main/scala/challenges/aoc2024/day5/Application.scala
package challenges.aoc2024.day5

import _root_.service.TaskSolver
import challenges.aoc2024.day5.service.{DayFiveSolver, PageProductionService}

object Application {
  def main(args: Array[String]): Unit = {
    val solver: TaskSolver = TaskSolver(DayFiveSolver(PageProductionService()))
    solver.solveTasks()
  }
}
